### PR TITLE
Feature/hocs 3896 business area extract correspondent

### PR DESCRIPTION
**Pull request by gardnersam**

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/caseworkclient/dto/GetCorrespondentWithPrimaryFlagResponse.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.digital.ho.hocs.audit.export.caseworkclient.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public class GetCorrespondentWithPrimaryFlagResponse {
```
Comment (robertdeniszczyc2) [inline src/main/java/uk/gov/digital/ho/hocs/audit/export/caseworkclient/dto/GetCorrespondentWithPrimaryFlagResponse.java:10]: Could the primary flag not be added to the existing GetCorrespondentOutlineResponse DTO?

File: src/main/java/uk/gov/digital/ho/hocs/audit/application/RequestData.java [hunk 2]
```diff
@@ -30,6 +31,7 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
         MDC.put(USER_ID_HEADER, initialiseUserId(request));
         MDC.put(USERNAME_HEADER, initialiseUserName(request));
         MDC.put(USER_ROLES_HEADER, initialiseUserRoles(request));
+        MDC.put(GROUP_HEADER, initialiseGroups(request));
```
Comment (robertdeniszczyc2) [inline src/main/java/uk/gov/digital/ho/hocs/audit/application/RequestData.java:34]: I'm not too familiar with this so may need someone else to review, why do we need to do this stuff with groups?

**Review by robertdeniszczyc2 — COMMENTED** [review]
Added a couple questions

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/caseworkclient/dto/GetCorrespondentWithPrimaryFlagResponse.java (hunk 1 above)
Comment (gardnersam) [reply src/main/java/uk/gov/digital/ho/hocs/audit/export/caseworkclient/dto/GetCorrespondentWithPrimaryFlagResponse.java]: Yep that's a good plan, I've updated to do this instead

**Pull request closed without merging**